[2025-09-20T11:52:44.543+0200] {taskinstance.py:1103} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: manual_full_load.load_promotions_staging manual__2025-09-20T09:52:40.324658+00:00 [queued]>
[2025-09-20T11:52:44.554+0200] {taskinstance.py:1103} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: manual_full_load.load_promotions_staging manual__2025-09-20T09:52:40.324658+00:00 [queued]>
[2025-09-20T11:52:44.555+0200] {taskinstance.py:1308} INFO - Starting attempt 1 of 2
[2025-09-20T11:52:44.573+0200] {taskinstance.py:1327} INFO - Executing <Task(PostgresOperator): load_promotions_staging> on 2025-09-20 09:52:40.324658+00:00
[2025-09-20T11:52:44.577+0200] {standard_task_runner.py:57} INFO - Started process 6228 to run task
[2025-09-20T11:52:44.580+0200] {standard_task_runner.py:84} INFO - Running: ['airflow', 'tasks', 'run', 'manual_full_load', 'load_promotions_staging', 'manual__2025-09-20T09:52:40.324658+00:00', '--job-id', '46', '--raw', '--subdir', 'DAGS_FOLDER/manual_full_load.py', '--cfg-path', '/tmp/tmpv1b9u33v']
[2025-09-20T11:52:44.584+0200] {standard_task_runner.py:85} INFO - Job 46: Subtask load_promotions_staging
[2025-09-20T11:52:44.651+0200] {task_command.py:410} INFO - Running <TaskInstance: manual_full_load.load_promotions_staging manual__2025-09-20T09:52:40.324658+00:00 [running]> on host 1102361e702e
[2025-09-20T11:52:44.746+0200] {taskinstance.py:1547} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='manual_full_load' AIRFLOW_CTX_TASK_ID='load_promotions_staging' AIRFLOW_CTX_EXECUTION_DATE='2025-09-20T09:52:40.324658+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-09-20T09:52:40.324658+00:00'
[2025-09-20T11:52:44.747+0200] {sql.py:265} INFO - Executing: 
        INSERT INTO staging.promotions_stg (promo_id, product_id, discount, start_date, end_date)
        SELECT promo_id, product_id, discount, start_date, end_date
        FROM raw.promotions;
        
[2025-09-20T11:52:44.767+0200] {base.py:73} INFO - Using connection ID 'postgres_retail' for task execution.
[2025-09-20T11:52:44.832+0200] {base.py:73} INFO - Using connection ID 'postgres_retail' for task execution.
[2025-09-20T11:52:44.838+0200] {sql.py:375} INFO - Running statement: 
        INSERT INTO staging.promotions_stg (promo_id, product_id, discount, start_date, end_date)
        SELECT promo_id, product_id, discount, start_date, end_date
        FROM raw.promotions;
        , parameters: None
[2025-09-20T11:52:44.839+0200] {taskinstance.py:1824} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/common/sql/operators/sql.py", line 277, in execute
    **extra_kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/common/sql/hooks/sql.py", line 349, in run
    self._run_command(cur, sql_statement, parameters)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/common/sql/hooks/sql.py", line 380, in _run_command
    cur.execute(sql_statement)
psycopg2.errors.UndefinedTable: relation "raw.promotions" does not exist
LINE 4:         FROM raw.promotions;
                     ^

[2025-09-20T11:52:44.852+0200] {taskinstance.py:1350} INFO - Marking task as UP_FOR_RETRY. dag_id=manual_full_load, task_id=load_promotions_staging, execution_date=20250920T095240, start_date=20250920T095244, end_date=20250920T095244
[2025-09-20T11:52:44.868+0200] {standard_task_runner.py:109} ERROR - Failed to execute job 46 for task load_promotions_staging (relation "raw.promotions" does not exist
LINE 4:         FROM raw.promotions;
                     ^
; 6228)
[2025-09-20T11:52:44.884+0200] {local_task_job_runner.py:225} INFO - Task exited with return code 1
[2025-09-20T11:52:44.942+0200] {taskinstance.py:2653} INFO - 0 downstream tasks scheduled from follow-on schedule check
